src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/DocumentDbEntityInformation.java

```diff
@@ -39,9 +39,7 @@ public DocumentDbEntityInformation(Class<T> domainClass) {
         super(domainClass);
 
         this.id = getIdField(domainClass);
-        if (this.id != null) {
-            ReflectionUtils.makeAccessible(this.id);
-        }
+        ReflectionUtils.makeAccessible(this.id);
```

Comment:
`this.id` will never be null?

Reply:
yeah, I change the getIdField somewhat like 
```java
if (idField == null) {
            throw new IllegalArgumentException("domain should contain @Id field or field named id");
````